### PR TITLE
Compare means to detect drift

### PR DESCRIPTION
**Pull request by SudipSinha**
# Add CompareMeans Drift Detection Metric

## Summary

This PR introduces a new `CompareMeans` drift detection metric that uses independent two-sample t-tests to detect distribution drift by comparing means between reference and current distributions. The implementation replaces the deprecated `Meanshift` endpoint with a more robust and well-tested solution.

### New Features
- **CompareMeans metric**: New drift detection using independent two-sample t-tests
  - Core implementation with support for Welch's and Student's t-tests
  - Full REST API with compute, definition, schedule, and list endpoints
  - Multi-feature drift detection support
  - Backwards compatible with deprecated `/metrics/drift/meanshift` endpoints

### Bug Fixes
- Fixed drift consistency: p-value now always matches drift_detected status in multi-feature scenarios
- Fixed type violations in `retrieve_tags()` method
- Fixed p-value selection in multivariate case

### Test Improvements
- Added comprehensive test coverage (10 core tests + full endpoint test suite)
- Refactored tests to use factory methods, removed redundant tests
- Added new factory methods for edge cases (small samples, different variances)
- Modernized type hints and code style

## Testing

All tests pass: **10 core metric tests + comprehensive endpoint test suite**
- Core tests cover drift detection, sample sizes, variance handling, edge cases
- Endpoint tests cover all API operations, error handling, and backwards compatibility

## Deprecation

**MeanShift endpoints are deprecated** (`/metrics/drift/meanshift/*`)

The "Meanshift" name was deprecated as it may be confused with the Mean Shift clustering algorithm. As reference, see
- [Mean shift - Wikipedia](https://en.wikipedia.org/wiki/Mean_shift)
- [MeanShift — scikit-learn documentation](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.MeanShift.html)
- [A demo of the mean-shift clustering algorithm — scikit-learn documentation](https://scikit-learn.org/stable/auto_examples/cluster/plot_mean_shift.html)

The new `CompareMeans` implementation provides:
- More accurate naming that clearly describes the metric's purpose
- Improved robustness and comprehensive test coverage
- Better error handling and consistency fixes

Old endpoints remain functional for backwards compatibility but will log deprecation warnings. Please migrate to `/metrics/drift/comparemeans/*` endpoints.

## API Usage

```json
POST /metrics/drift/comparemeans
{
  "modelId": "my-model",
  "referenceTag": "baseline",
  "fitColumns": ["feature1", "feature2"],
  "alpha": 0.05,
  "equalVar": false,
  "nanPolicy": "omit"
}
```

## Summary by Sourcery

Introduce a new CompareMeans drift metric based on independent two-sample t-tests, wire it into the FastAPI application alongside Prometheus/CORS improvements, and remove the legacy Meanshift implementation while keeping deprecated endpoints as proxies for backward compatibility.

New Features:
- Add a CompareMeans drift detection core metric using independent two-sample t-tests with configurable alpha, variance assumption, and NaN handling.
- Expose full REST endpoints for CompareMeans (compute, definition, schedule, delete, list) and register them with OpenAPI and application routing.
- Add compatibility handlers so deprecated Meanshift endpoints delegate to the new CompareMeans implementation while being marked deprecated in the API spec.

Bug Fixes:
- Ensure multi-feature drift responses use p-values from drifting features so drift_detected is always consistent with the returned p-value.
- Fix type-safety and defaulting for metric request tags so Prometheus tagging no longer violates Dict[str, str] expectations.
- Correct handling of empty data and invalid IDs across drift metric endpoints with clearer HTTP error responses.

Enhancements:
- Expand and restructure drift metric core test factories to cover small samples, unequal variances, parameter independence, and multivariate behavior.
- Add comprehensive endpoint tests for CompareMeans, deprecated Meanshift proxies, scheduler interactions, and OpenAPI registration.
- Improve FastAPI app typing, Prometheus metrics endpoint behavior, and CORS configuration while refining Hypercorn keep-alive configuration.
- Refine KSTest endpoint defaults and generalize shared endpoint test factories for clearer success, error, edge-case, and exception handling coverage.

Build:
- Relax lower bounds for prometheus-client and pydantic dependencies and reformat dependency groups in pyproject configuration.

CI:
- Update pre-commit hook versions for core linters and security tools including flake8, ruff, gitleaks, and mypy.

Tests:
- Add core metric tests for CompareMeans including regression checks against SciPy and parameter-specific behavior for equal_var and NaN policy.
- Add integration tests ensuring CompareMeans and deprecated Meanshift endpoints appear correctly in OpenAPI, have proper tags, and that Prometheus and CORS behavior is validated.

### CODE REVIEW

File: src/endpoints/metrics/drift/compare_means.py [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+    return await schedule_CompareMeans(compare_means_request)
+
+
+@router.delete("/metrics/drift/meanshift/request", deprecated=True)
+async def delete_meanshift_schedule(schedule: ScheduleId):
+    """Delete a recurring computation of Meanshift metric (deprecated).
+
+    This endpoint is deprecated. Please use /metrics/drift/comparemeans/request instead.
+    """
+    logger.warning(f"Deprecated {DEPRECATED_METRIC_NAME} endpoint called. Use {METRIC_NAME} endpoint instead.")
```
Comment (RobGeada) [inline src/endpoints/metrics/drift/compare_means.py]: I have a hunch we're gonna need this warning in a lot more places in the future- can we create a common function to log a deprecation warning?

Comment (SudipSinha) [reply src/endpoints/metrics/drift/compare_means.py]: Yeah, good idea! Let me get to that.

Comment (SudipSinha) [reply src/endpoints/metrics/drift/compare_means.py]: Addressed in d9dc9aaa5f1802fe1045845dd3e02867355d7b75.

**Review by RobGeada — APPROVED** [review]
